www/js/metrics/MetricsDateSelect.tsx

```diff
@@ -35,11 +35,24 @@ const MetricsDateSelect = ({ dateRange, setDateRange }: Props) => {
 
   const onChoose = useCallback(
     ({ startDate, endDate }) => {
+      const dtStartDate = DateTime.fromJSDate(startDate).startOf('day');
+      let dtEndDate;
+
+      if (!endDate) {
+        // If no end date selected, pull range from then till present day
+        dtEndDate = DateTime.now();
+      } else if (
+        dtStartDate.toString() === DateTime.fromJSDate(endDate).startOf('day').toString()
+      ) {
+        // For when only one day is selected
+        // NOTE: As written, this technically timestamp will technically fetch _two_ days.
```

Comment:
@the-bay-kay can likely provide better insight than I can

Reply:
Sure thing! This comment was a result of the work we did to fix the White-Screen of Death showstopper on the Metrics tab.  All the notes are in [this](https://github.com/JGreenlee/e-mission-phone/pull/1) PR, if you want to read more!

As written, selecting a single day (endDate being the end of the startDate), actually fetches _two_ days worth of metrics from the server.  When we query for any shorter of a range (endDate < the end of the startDate), _no_ days are fetched, and the metrics tab returns empty.   (Put another way: if you select 1/17/24, the metrics display would be identical to those displayed in the range 1/17/24-1/18/24)/

I spent a fair bit of time digging through the server, tracing the call all the way back to the DB Query, and couldn't figure out why the query resulted in two days.  Furthermore, "open-ended date selection" seems to be an issue with the date-picker plugin we use; on the JS Playground example they give, I noticed that choosing open ended ranges resulted in undefined behavior.  I think there's future work to be done (a) sending a PR to the date picker, to change the open-ended behavior, and (b) debugging the server queries to figure out why single-day metrics aren't available.

The actual WSOD was an issue with this query; because the end date was undefined, errors were thrown and the page would crash.  After fixing this, we had a few options:
- Make single day queries appear blank
- Make single day queries show the "incorrect" 48-hour period (Which, when selecting the present day, _is_ actually correct!)
- Detect single day selections,and throw a pop-up warning prompting the user to re-select a range of 2+ days.
    - Unfortunately, the Date-Picker plugin does not support this prevention natively.  Ideally, we'd stop the user from being able to select a single day, but that'd require a pretty hefty rework of the plugin).

Because our priority was getting this done and dusted for staging, Jack and I agreed the best option was to display the "two-day" period, while documenting the behavior.  This felt like the most user-friendly solution; since the metrics tab already shows ballpark values, we figured having it not be _exactly_ the day was sufficient for now.  I'd like to wrap back around and fix this once we have time, but the priority was fixing the WSOD error!